qf-faucet-server/src/main.rs

```diff
@@ -0,0 +1,302 @@
+use std::str::FromStr;
+
+use subxt;
+use subxt::{OnlineClient, PolkadotConfig};
+use subxt_signer::bip39::Mnemonic;
+use warp::{reject::Reject, Filter};
+use subxt_signer::sr25519::dev;
+use subxt_signer::sr25519;
+use std::env;
+use hex;
+use clap::Parser;
+use subxt::utils::{MultiAddress, AccountId32};
+use rusqlite::{Connection, Result as SqlResult};
+use chrono::Utc;
+use serde::{Deserialize, Serialize};
+
+
+#[subxt::subxt(runtime_metadata_path = "metadata.scale", derive_for_all_types = "PartialEq, Eq")]
+pub mod polkadot {}
+
+const RPC_URL: &str = "wss://dev.qfnetwork.xyz/socket";
+const FAUCET_AMOUNT: u128 = 20_000_000_000;
+
+#[derive(Debug, Deserialize, Serialize)]
+struct DripRequest {
+    address: String,
+}
+
+#[derive(Debug)]
+enum Errors {
+    TransferError,
+    StorageError,
+    SomeError,
+}
+
+
+impl Reject for Errors {}
+
+fn init_db() {
+    let conn = get_db().expect("Init DB Error");
+    conn.execute(
+        "CREATE TABLE IF NOT EXISTS transfers (
+            id INTEGER PRIMARY KEY,
+            address TEXT NOT NULL,
+            amount INTEGER NOT NULL,
+            timestamp INTEGER NOT NULL,
+            tx_hash TEXT NOT NULL
+        )",
+        [],
+    ).expect("Create Table Error");
+    
+    // Add index on address column
+    conn.execute(
+        "CREATE INDEX IF NOT EXISTS idx_transfers_address ON transfers (address)",
+        [],
+    ).expect("Create Index Error");
+    
+    // Add index on timestamp column
+    conn.execute(
+        "CREATE INDEX IF NOT EXISTS idx_transfers_timestamp ON transfers (timestamp)",
+        [],
+    ).expect("Create Index Error");
+    
+    // Add composite index on both address and timestamp for the specific query pattern
+    conn.execute(
+        "CREATE INDEX IF NOT EXISTS idx_transfers_address_timestamp ON transfers (address, timestamp)",
+        [],
+    ).expect("Create Index Error");
+}
+
+fn get_db() -> SqlResult<Connection> {
+    let conn = Connection::open("transfers.db")?;
+    Ok(conn)
+}
+
+fn store_transfer(address: &str, amount: u64, tx_hash: &str, time: i64) -> SqlResult<()> {
+    let conn = get_db()?;
+    conn.execute(
+        "INSERT INTO transfers (address, amount, timestamp, tx_hash) VALUES (?1, ?2, ?3, ?4)",
+        (address, amount, time, tx_hash),
+    )?;
+    Ok(())
+}
+
+fn update_transfer(address: &str, time: i64, tx_hash: &str) -> SqlResult<()> {
+    let conn = get_db()?;
+    conn.execute(
+        "UPDATE transfers SET tx_hash = ?2 WHERE address = ?1 AND timestamp = ?3",
+        (address, tx_hash, time),
+    )?;
+    Ok(())
+}
+
+fn can_transfer(address: &str, timeout: Option<u64>) -> SqlResult<bool> {
+    let conn = get_db()?;
+    let now = Utc::now().timestamp() as u64;
+    let two_hours_ago = now - (timeout.unwrap_or_else(|| 120) * 60);
+
+    // Check if there are recent transfers
+    let count: i64 = conn.query_row(
+        "SELECT COUNT(*) FROM transfers 
```

Comment:
Should also work with `usize`:

https://docs.rs/rusqlite/0.33.0/rusqlite/types/trait.FromSql.html#impl-FromSql-for-usize

```suggestion
    let count: usize = conn.query_row(
        "SELECT COUNT(*) FROM transfers 
```

Reply:
for what?.. It is not an error and all working fine